### ISSUE TITLE
New Arduino functions wrapping Analog/DigitalRead, DigitalWrite, I2C, SPI (Trac #2)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#6]
Simple wrapper functions to remotely control I/O on the Arduino pins. These functions should map to the Arduino API as closely as possible. We can probably replace the current SampleVoltage call with AnalogRead (or an overloaded variant supporting multiple samples).

Migrated from http://microfluidics.utoronto.ca/ticket/2
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "Simple wrapper functions to remotely control I/O on the Arduino pins. These functions should map to the Arduino API as closely as possible. We can probably replace the current SampleVoltage call with AnalogRead (or an overloaded variant supporting multiple samples).", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "New Arduino functions wrapping Analog/DigitalRead, DigitalWrite, I2C, SPI", 
    "priority": "major", 
    "keywords": "", 
    "version": "", 
    "time": "2011-09-18T05:06:58", 
    "milestone": "Microdrop 1.0", 
    "owner": "ryan", 
    "type": "enhancement"
}
```

**Comment by cfobel** [comment]
Trac update at 20110919T07:10:42: ryan changed description from "Simple wrapper functions to remotely control IO on Arduino pins." to "Simple wrapper functions to remotely control I/O on the Arduino pins. These functions should map to the Arduino API as closely as possible. We can probably replace the current SampleVoltage call with AnalogRead (or an overloaded variant supporting multiple samples)."

**Comment by cfobel** [comment]
Trac update at 20111028T17:02:22: 
* ryan@fobel.net commented:

> commit ce2946bcf0da6e282a854292433dd4e397b8b5e4
> Author: Ryan Fobel <ryan@fobel.net>
> Date:   Fri Oct 28 12:56:39 2011 -0400
> 
>     Add i2c_read command to dmf_control_board
>     
>     Closes #6.


* ryan@fobel.net changed resolution from "" to "fixed"
* ryan@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "hardware/communication" to "microdrop"
